src/software/thunderscope/binary_context_managers.py

```diff
@@ -63,7 +66,7 @@ def __init__(
 
         self.thread = threading.Thread(target=self.__restart__)
 
-    def __enter__(self):
+    def __enter__(self) -> "self":
```

Comment:
I think you could return `Self` (capital S) here instead of a string: https://realpython.com/python-type-self/

Reply:
We're not on python 3.11 yet so we I think we can't do this; alternative would be to `from __future__ import annotations` and then you can use the actual type